### DISCUSSION TITLE
what is the size limit of duckdb in wasm?

### DISCUSSION THREAD

**Discussion by kesavkolla** [duckdb/duckdb-wasm#1241] [Q&A]
Is there any restriction on the size of the database in wasm?

**Comment by carlopi** [comment]
Currently memory is constrained by WebAssembly 32 bit pointers, so short of 2GB.

There is a potential solution to this in the mid-term, but nothing that is currently worked on.

**Reply by mullinsms** [reply]
I thought [WebAssembly added support for up to 4GB](https://v8.dev/blog/4gb-wasm-memory) two years ago. Is it not an option to explicitly opt in to using more memory?

**Reply by bendnorman** [reply]
Can the size of the database only be 2 GB, or does duckdb WASM only have 2GB of memory to run queries? Is there an actual limit to the size of a parquet file you analyze in duckdb wasm?

**Reply by ravwojdyla** [reply]
Linking https://github.com/duckdb/duckdb-wasm/issues/1705

**Comment by kesavkolla** [comment]
@carlopi So currently all duckdb wasm usage is all about reading remote data via parquet/csv/json files. There is nothing that can be saved on client side is that correct?

**Comment by zackify** [comment]
It seems wasm supports 64bit now in a few browsers? What needs to be done to utilize this with duckdb for larger databases client side.

**Reply by arghaffari** [reply]
@carlopi
